pkg/kubectl/cmd/apply.go

```diff
@@ -373,12 +376,36 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 		}
 		count++
 		if printObject {
-			return cmdutil.PrintObject(cmd, info.Object, out)
+			objs = append(objs, info.Object)
+			return nil
 		}
 		cmdutil.PrintSuccess(shortOutput, out, info.Object, dryRun, "configured")
 		return nil
 	})
 
+	if len(objs) > 0 {
```

Comment:
I think this should be after L411.

Because `len(objs)` could be larger than 1, when the `err` from `r.Visit` is non-nil. Because the error could be returned from a child visitor, which will terminate iteration. 

WDYT?

Reply:
Thanks, updated.